report_builder/models.py

```diff
@@ -85,18 +88,18 @@ def save(self, *args, **kwargs):
             unique_slugify(self, self.name)
         super(Report, self).save(*args, **kwargs)
 
-    def add_aggregates(self, queryset):
-        for display_field in self.displayfield_set.filter(aggregate__isnull=False):
-            if display_field.aggregate == "Avg":
-                queryset = queryset.annotate(Avg(display_field.path + display_field.field))
-            elif display_field.aggregate == "Max":
-                queryset = queryset.annotate(Max(display_field.path + display_field.field))
-            elif display_field.aggregate == "Min":
-                queryset = queryset.annotate(Min(display_field.path + display_field.field))
-            elif display_field.aggregate == "Count":
-                queryset = queryset.annotate(Count(display_field.path + display_field.field))
-            elif display_field.aggregate == "Sum":
-                queryset = queryset.annotate(Sum(display_field.path + display_field.field))
+    def add_aggregates(self, queryset, display_fields=None):
```

Comment:
Its unclear if this method mutates `self` or just returns an updated queryset. If it doesn't modify `self` than I would reccomend making this a method on a custom manager instead of a method on the model: https://docs.djangoproject.com/en/1.8/topics/db/managers/#modifying-initial-manager-querysets